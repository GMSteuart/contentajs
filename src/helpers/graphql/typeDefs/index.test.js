const typeDefs = require('./index');

describe('typeDefs', () => {
  // I know this is silly... but I'd like to have 100% coverage.
  it('is what we expect', async () => {
    const schema = await Promise.all(typeDefs);
    expect(schema).toMatchSnapshot();
  });
});
