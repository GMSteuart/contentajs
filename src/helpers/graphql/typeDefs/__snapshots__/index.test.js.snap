// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`typeDefs is what we expect 1`] = `
Array [
  "type Article {
  id: String!
  title: String!
  owner: User!
}
",
  "type Query {
  lastRecipe: Recipe
    @fromJsonApi(
      query: \\"/recipes?page[limit]=1&sort=createdAt&include=author\\"
    )
  recipesByAuthor(authorName: String!): [Recipe]
    @fromJsonApi(
      query: \\"/recipes?filter[author.name]={authorName}&include=author\\"
    )
  articlesByAuthor(authorName: String!): [Article]
    @fromJsonApi(
      query: \\"/articles?filter[owner.name]={authorName}&include=owner\\"
    )
}
",
  "type Recipe {
  id: String!
  title: String!
  ingredients: [String!]
  instructions: String!
  preparationTime: Int
  author: User!
}
",
  "type User {
  id: String!
  name: String!
  recipes: [Recipe]
}
",
]
`;
